libs/external-router/routing/src/guard/external-matcher-type.guard.ts

```diff
@@ -0,0 +1,73 @@
+import { inject } from '@angular/core';
+import {
+  CanMatchFn,
+  Router,
+  UrlTree,
+} from '@angular/router';
+import {
+  Observable,
+  catchError,
+  map,
+  of,
+  tap,
+} from 'rxjs';
+
+import {
+  DAFF_EXTERNAL_ROUTER_CONFIG,
+  DaffExternalRouterConfiguration,
+  DaffExternallyResolvableUrl,
+  DaffRouteWithSeoData,
+} from '@daffodil/external-router';
+import { DaffExternalRouterDriver } from '@daffodil/external-router/driver';
+
+import { DaffExternalRouterNotFoundError } from '../errors/not-found-error';
+import { DaffExternalRouterPermanentRedirectError } from '../errors/permanent-redirect';
+import { DaffExternalRouterTemporaryRedirectError } from '../errors/temporary-redirect';
+import { daffConvertToPath } from '../helper/convert-to-path';
+import { processErrors } from '../processors/process-errors';
+import { processRedirects } from '../processors/process-redirect';
+
+/**
+ * A `canMatch` guard function that will resolve a route externally, allowing Angular to match it.
+ *
+ * Depending on the resolution result, the route's data will contain `daffSeoData` with supplemental search engine data
+ * about the route. The data on the route is cumulative. If other data properties are set, they will be merged alongside daffSeoData.
+ */
+export const daffExternalMatcherTypeGuard = (type: string) => (route: DaffRouteWithSeoData, segments): Observable<boolean | UrlTree> => {
+  const router = inject(Router);
+  const config: DaffExternalRouterConfiguration = inject(DAFF_EXTERNAL_ROUTER_CONFIG);
+  return inject(DaffExternalRouterDriver).resolve(daffConvertToPath(segments)).pipe(
+    map((resolvedRoute: DaffExternallyResolvableUrl) => processErrors(resolvedRoute)),
+    map((resolvedRoute: DaffExternallyResolvableUrl) => processRedirects(resolvedRoute)),
```

Comment:
```suggestion
    map(processErrors),
    map(processRedirects),
```

I much prefer this functional style

Reply:
Agreed, fixed in 51703a556